### PR TITLE
Bug 1841171: Update PkgManifest upon catsrc update

### PR DESCRIPTION
**Pull request by Bowenislandsong**
The catsrc update sync triggered by the informer does not update pkgmanifest. This commit adds an update mechanism.

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Comment by kevinrizza** [comment]
/approve
lgtm

File: test/e2e/packagemanifest_e2e_test.go [hunk 1]
```diff
@@ -17,15 +17,222 @@ import (
...
+var _ = Describe("Package Manifest API lists available Operators from Catalog Sources", func() {
+
+	var (
+		crc versioned.Interface
+		pmc pmversioned.Interface
+		c   operatorclient.ClientInterface
+	)
+	BeforeEach(func() {
+		crc = newCRClient()
+		pmc = newPMClient()
```
Comment (njhale) [inline test/e2e/packagemanifest_e2e_test.go:29]: Whoops -- these calls are missing some arguments!

**Comment by njhale** [comment]
/hold

just until the test is fixed.

File: test/e2e/packagemanifest_e2e_test.go (hunk 1 above)
Comment (Bowenislandsong) [reply test/e2e/packagemanifest_e2e_test.go:29]: they are?

**Comment by Bowenislandsong** [comment]
/hold cancel 
Test fixed

**Pull request merged**